managementAgent.js

```diff
@@ -1,51 +1,130 @@
+                    throw new Error('failed to init management');
+                }
+            });
+        });
+    }
+
+    stop() {
+        if (this.wss) {
+            this.wss.close(() => {
+                logger.info('server shutdown');
+            });
+        }
+    }
+
+    startServer() {
+        this.wss = new WebSocket.Server({
+            port: this.port,
+            clientTracking: true,
+            path: '/watch'
+        });
 
-process.env.REPORT_TOKEN = process.env.REPORT_TOKEN
-                           || _config.reportToken
-                           || Uuid.v4();
+        this.wss.on('connection', this.onConnection.bind(this));
+        this.wss.on('listening', this.onListening.bind(this));
+        this.wss.on('error', this.onError.bind(this));
 
-setTimeout(() => {
-    initManagement(logger.newRequestLogger());
-}, 5000);
+        setInterval(this.checkBrokenConnections.bind(this),
+                    CHECK_BROKEN_CONNECTIONS_FREQUENCY_MS);
 
-setTimeout(() => {
-    managementAgentWS();
-}, 6);
+        addOverlayMessageListener(this.onNewOverlay.bind(this));
+    }
 
-function managementAgentWS() {
-    const port = _config.managementAgent.port || 8010;
-    const wss = new WebSocket.Server({
-        port: port,
-        clientTracking: true,
-        path: '/watch'
-    });
+    onConnection(socket, request) {
+        function hearthbeat() {
+            this.isAlive = true;
+        }
 
-    wss.on('connection', ws => {
-        logger.info('management agent: client connected to watch route');
-    });
+        logger.info('client connected to watch route', {
+            ip: request.connection.remoteAddress,
+        });
 
-    wss.on('listening', ws => {
-        logger.info('management agent websocket server listening', { port });
-    });
+        socket.isAlive = true;
+        socket.on('pong', hearthbeat.bind(socket));
 
-    wss.on('error', error => {
-        logger.error('management agent websocket server error', { error });
-    });
+        if (!socket.readyState === socket.OPEN) {
+            logger.error('client socket not in ready state', { socket });
+            return;
+        }
 
-    addOverlayMessageListener((remoteOverlay) => {
-        wss.clients.forEach((client) => {
+        const msg = {
+            messageType: managementAgentMessageType.LOADED_OVERLAY,
+            payload: this.loadedOverlay
+        };
+        socket.send(JSON.stringify(msg), (error) => {
+            if (error) {
+                logger.error('failed to send remoteOverlay to client',
+                             { error, socket });
+            }
+        });
+    }
+
+    onListening() {
+        logger.info('websocket server listening',
+                    { port: this.port });
+    }
+
+    onError(error) {
+        logger.error('websocket server error', { error });
+    }
+
+    onNewOverlay(remoteOverlay) {
+        //XXX patch configuration?
```

Comment:
I don't think the server has anything to patch as the overlay only contains s3-protocol specific items. On thing that will come later is the `browserAccess` flag